### PR TITLE
[OldDot Rules Migration] Add "Rules" Row/toggle to "More Features" page

### PR DESCRIPTION
**Pull request by WojtekBoman**
**Holding on production deploy:**

- [x] [Web-Expensify/pull/43192](https://github.com/Expensify/Web-Expensify/pull/43192)

### Details

This PR adds a button to toggle rules on the More Features screen. Now the rules screen only contains empty sections that will be added in the next PRs

### Fixed Issues

$ https://github.com/Expensify/App/issues/46935
PROPOSAL:

### Tests

Preconditions:
Make sure you have access to the `workspaceRules` beta. If you run the app locally you can modify the `canUseWorkspaceRules` or `canUseAllBetas` function in `Permissions.ts` to always return `true`.

- [ ] Verify that no errors appear in the JS console

1. Create a new workspace (from the Workspace switcher or the Settings/Workspaces screen).
2. Go to the More features screen.
3. Enable the Rules.
4. Upgrade your workspace if it's necessary.
5. The rules screen should now be available.

### QA Steps

1. With an account that is not on the beta verify these new features are not accessible.
2. With an account that is on the beta verify steps 1-5 from the Tests section.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/2980e0d1-cfd6-427b-b6da-b5f4f0d6ce4c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/c73485da-859a-4cce-863b-73c864847bed

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/3c9898ce-3bd8-428e-bfb3-1cf0e39e67a4

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/a0ef43f1-4cb8-4e8b-aa18-07baa25caa69

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/74a2baf2-06a3-4872-8636-c9c291b8165f

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/b75b84fc-f293-4106-a1fa-c7c4c9f87464

</details>

### CODE REVIEW

File: src/languages/es.ts
Comment (WojtekBoman) [inline src/languages/es.ts:1]: @JmillsExpensify These translations were generated using chatGPT (https://chatgpt.com/g/g-tAon5M5jH-english-to-spanish-translator), could someone review them?

File: src/CONST.ts [hunk 1]
```diff
@@ -86,6 +86,9 @@ const CONST = {
     DEFAULT_TABLE_NAME: 'keyvaluepairs',
     DEFAULT_ONYX_DUMP_FILE_NAME: 'onyx-state.txt',
     DEFAULT_POLICY_ROOM_CHAT_TYPES: [chatTypes.POLICY_ADMINS, chatTypes.POLICY_ANNOUNCE, chatTypes.DOMAIN_ALL],
+    DEFAULT_MAX_EXPENSE_AMOUNT_NO_RECEIPT: 10000000000,
+    DEFAULT_MAX_EXPENSE_AMOUNT: 10000000000,
+    DEFAULT_MAX_EXPENSE_AGE: 10000000000,
```
Comment (BrtqKr) [inline src/CONST.ts:91]: isn't it the same as the ones from policy section of constants

```
        DEFAULT_MAX_EXPENSE_AGE: 90,
        DEFAULT_MAX_EXPENSE_AMOUNT: 200000,
        DEFAULT_MAX_AMOUNT_NO_RECEIPT: 2500,
```

File: src/CONST.ts [hunk 2]
```diff
@@ -5459,6 +5464,14 @@ const CONST = {
                 description: 'workspace.upgrade.taxCodes.description' as const,
                 icon: 'Coins',
             },
+            rules: {
+                id: 'rules' as const,
+                alias: 'rules',
+                name: 'Rules',
+                title: 'workspace.upgrade.rules.title' as const,
+                description: 'workspace.upgrade.rules.description' as const,
```
Comment (BrtqKr) [inline src/CONST.ts:5472]: 1. Do we need as const in those fields?
2. If yes, why wouldn't we use as const for the whole array/object

**Review by BrtqKr — APPROVED** [review]
Minor things to clarify, but overall seems fine

File: src/CONST.ts (hunk 1 above)
Comment (WojtekBoman) [reply src/CONST.ts]: I renamed the prefix of the added consts from `DEFAULT_` to `EMPTY_`

File: src/CONST.ts (hunk 2 above)
Comment (WojtekBoman) [reply src/CONST.ts:5472]: It has been implemented according to the other consts in this getter

**Comment by shawnborton** [comment]
When you go to upgrade your policy after tapping on the Rules toggle, I would think that the rules toggle should be toggled on automatically as soon as you upgrade. But right now it seems like you have to press the toggle a second time after the upgrade is complete.

**Comment by dannymcclain** [comment]
>When you go to upgrade your policy after tapping on the Rules toggle, I would think that the rules toggle should be toggled on automatically as soon as you upgrade.

Totally agree—was just about to make this comment!

The rules toggle is otherwise looking good to me.

**Comment by WojtekBoman** [comment]
> When you go to upgrade your policy after tapping on the Rules toggle, I would think that the rules toggle should be toggled on automatically as soon as you upgrade. But right now it seems like you have to press the toggle a second time after the upgrade is complete.

Thanks for noticing that! It has already been fixed :)

https://github.com/user-attachments/assets/40d7234c-a10c-4e73-97d8-c963730d3e53


https://github.com/user-attachments/assets/a90937dd-4b9c-4972-a34d-0983ca029055

File: src/libs/API/types.ts [hunk 3]
```diff
@@ -548,6 +550,7 @@ type WriteCommandParameters = {
     [WRITE_COMMANDS.REQUEST_EXPENSIFY_CARD_LIMIT_INCREASE]: Parameters.RequestExpensifyCardLimitIncreaseParams;
     [WRITE_COMMANDS.CLEAR_OUTSTANDING_BALANCE]: null;
     [WRITE_COMMANDS.CANCEL_BILLING_SUBSCRIPTION]: Parameters.CancelBillingSubscriptionParams;
+
```
Comment (marcaaron) [inline src/libs/API/types.ts:553]:
```suggestion
```

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx [hunk 4]
```diff
@@ -126,6 +113,39 @@ function WorkspaceMoreFeaturesPage({policy, route}: WorkspaceMoreFeaturesPagePro
...
+        {
+            icon: Illustrations.Rules,
+            titleTranslationKey: 'workspace.moreFeatures.rules.title',
+            subtitleTranslationKey: 'workspace.moreFeatures.rules.subtitle',
+            isActive: policy?.areRulesEnabled ?? false,
+            pendingAction: policy?.pendingFields?.areRulesEnabled,
+            action: (isEnabled: boolean) => {
+                if (!policyID) {
+                    return;
+                }
```
Comment (marcaaron) [inline src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:139]: Can we reach this without a policyID? 🤔

File: src/languages/en.ts [hunk 5]
```diff
@@ -3546,6 +3560,16 @@ export default {
             chatInAdmins: 'Chat in #admins',
             addPaymentCard: 'Add payment card',
         },
+        rules: {
+            individualExpenseRules: {
+                title: 'Expenses',
+                subtitle: 'Set spend controls and defaults for individual expenses. You can also create rules for categories and tags.',
```
Comment (marcaaron) [inline src/languages/en.ts:3566]: Did we want "categories" and "tags" to be links?

**Review by marcaaron — COMMENTED** [review]
Looks good so far. Had a few small notes. Gonna do some testing.

File: src/languages/es.ts
Comment (marcaaron) [reply src/languages/es.ts:1]: Yes, I will compile everything in this PR and request a review now.

File: src/pages/workspace/upgrade/WorkspaceUpgradePage.tsx [hunk 6]
```diff
@@ -47,6 +47,9 @@ function WorkspaceUpgradePage({route}: WorkspaceUpgradePageProps) {
             case CONST.UPGRADE_FEATURE_INTRO_MAPPING.reportFields.id:
                 Policy.enablePolicyReportFields(policyID, true, true);
                 return Navigation.navigate(ROUTES.WORKSPACE_MORE_FEATURES.getRoute(policyID));
+            case CONST.UPGRADE_FEATURE_INTRO_MAPPING.rules.id:
+                Policy.enablePolicyRules(policyID, true, true);
+                return Navigation.navigate(ROUTES.WORKSPACE_MORE_FEATURES.getRoute(policyID));
```
Comment (marcaaron) [inline src/pages/workspace/upgrade/WorkspaceUpgradePage.tsx:52]: Hmm this is violating the 1:1:1 principle now though... if we want this it should happen inside `upgradeToCorporate` and would need to be a backend change.

Comment (marcaaron) [reply src/pages/workspace/upgrade/WorkspaceUpgradePage.tsx:52]: Oh maybe not because there is a "Got it, thanks" screen 😄  nvm

**Comment by marcaaron** [comment]
In terms of roll out, it might be good to have at least one setting in these cards? Does this look too much like a bug?



Another option could be to use a beta. But I think it's probably fine to incrementally add more rules as we go.

What do you guys think?

**Comment by shawnborton** [comment]
I don't have strong feelings, though I do agree that exposing that screen to the user feels like a broken experience. I'm cool with putting something like this behind a beta. Thoughts @Expensify/design @JmillsExpensify @trjExpensify ?

**Comment by dannymcclain** [comment]
I feel the same as Shawn. Not _super_ passionate, but it does feel kinda `:womp:` to land at a screen that you really can't do anything with.

**Comment by trjExpensify** [comment]
I’d vote to put it behind a beta until we’ve built it out and ready to
release.

**Comment by dubielzyk-expensify** [comment]
Sounds good to me too 👍

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx (hunk 4 above)
Comment (WojtekBoman) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:143]: It's necessary to have this value defined when we invoke an action method, without this check TS throws an error

**Comment by WojtekBoman** [comment]
I hid this feature behind the `rules` beta, if this beta should have a different name, I'll adjust it

File: src/languages/en.ts (hunk 5 above)
Comment (WojtekBoman) [reply src/languages/en.ts]: Yes, these should be links :) I've already fixed that

https://github.com/user-attachments/assets/518f595c-12a6-46c5-ae5f-f094ec425daf

https://github.com/user-attachments/assets/b5917818-b1ba-4769-94c9-651aeca4ec53

File: src/CONST.ts [hunk 7]
```diff
@@ -381,6 +382,7 @@ const CONST = {
         REPORT_FIELDS_FEATURE: 'reportFieldsFeature',
         WORKSPACE_FEEDS: 'workspaceFeeds',
         NETSUITE_USA_TAX: 'netsuiteUsaTax',
+        RULES: 'rules',
```
Comment (marcaaron) [inline src/CONST.ts:385]:
```suggestion
        WORKSPACE_RULES: 'workspaceRules',
```

File: src/libs/Permissions.ts [hunk 8]
```diff
@@ -36,6 +36,10 @@ function canUseNetSuiteUSATax(betas: OnyxEntry<Beta[]>): boolean {
     return !!betas?.includes(CONST.BETAS.NETSUITE_USA_TAX) || canUseAllBetas(betas);
 }
 
+function canUseRules(betas: OnyxEntry<Beta[]>): boolean {
+    return !!betas?.includes(CONST.BETAS.RULES) || canUseAllBetas(betas);
+}
```
Comment (marcaaron) [inline src/libs/Permissions.ts:41]:
```suggestion
function canUseWorkspaceRules(betas: OnyxEntry<Beta[]>): boolean {
    return !!betas?.includes(CONST.BETAS.WORKSPACE_RULES) || canUseAllBetas(betas);
}
```

File: src/libs/Permissions.ts [hunk 9]
```diff
@@ -52,4 +56,5 @@ export default {
     canUseSpotnanaTravel,
     canUseWorkspaceFeeds,
     canUseNetSuiteUSATax,
+    canUseRules,
```
Comment (marcaaron) [inline src/libs/Permissions.ts:59]:
```suggestion
    canUseWorkspaceRules,
```

File: src/pages/workspace/WorkspaceInitialPage.tsx [hunk 10]
```diff
@@ -306,6 +310,15 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, reimbursementAcc
         });
     }
 
+    if (featureStates?.[CONST.POLICY.MORE_FEATURES.ARE_RULES_ENABLED] && canUseRules) {
```
Comment (marcaaron) [inline src/pages/workspace/WorkspaceInitialPage.tsx:313]:
```suggestion
    if (featureStates?.[CONST.POLICY.MORE_FEATURES.ARE_RULES_ENABLED] && canUseWorkspaceRules) {
```

File: src/pages/workspace/WorkspaceInitialPage.tsx [hunk 11]
```diff
@@ -99,6 +101,7 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, reimbursementAcc
     const activeRoute = useNavigationState(getTopmostRouteName);
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
+    const {canUseRules} = usePermissions();
```
Comment (marcaaron) [inline src/pages/workspace/WorkspaceInitialPage.tsx:104]:
```suggestion
    const {canUseWorkspaceRules} = usePermissions();
```

**Review by marcaaron — CHANGES_REQUESTED** [review]
Still waiting for translations to be reviewed internall. But everything looks good here! 

I renamed the `rules` beta to `workspaceRules`. We will have to wait some time for the beta changes to reach production.

File: src/languages/es.ts [hunk 12]
```diff
@@ -3621,6 +3635,16 @@ export default {
...
             addPaymentCard: 'Agregar tarjeta de pago',
         },
+        rules: {
+            individualExpenseRules: {
+                title: 'Gastos',
+                subtitle: 'Establece controles y valores predeterminados para gastos individuales. También puedes crear reglas para',
+            },
+            expenseReportRules: {
+                title: 'Informes de gastos',
+                subtitle: 'Automatiza el cumplimiento de los informes de gastos, las aprobaciones y el pago.',
```
Comment (pecanoro) [inline src/languages/es.ts:3645]:
```suggestion
                subtitle: 'Automatiza el cumplimiento, la aprobación y el pago de los informes de gastos',
```

File: src/languages/es.ts [hunk 13]
```diff
@@ -3590,6 +3599,11 @@ export default {
                 description: `Añada código de impuesto mayor a sus categorías para exportar fácilmente los gastos a sus sistemas de contabilidad y nómina.`,
                 onlyAvailableOnPlan: 'Los código de impuesto mayor solo están disponibles en el plan Control, a partir de ',
             },
+            rules: {
+                title: 'Reglas',
+                description: `Las reglas se ejecutan en segundo plano y mantienen tus gastos bajo control para que no tengas que preocuparte por los detalles pequeños.\n\nExige detalles de gastos como recibos y descripciones, establece límites y valores predeterminados, y automatiza aprobaciones y pagos, todo en un solo lugar.`,
```
Comment (pecanoro) [inline src/languages/es.ts:3604]:
```suggestion
                description: `Las reglas se ejecutan en segundo plano y mantienen tus gastos bajo control para que no tengas que preocuparte por los detalles pequeños.\n\nExige detalles de los gastos, como recibos y descripciones, establece límites y valores predeterminados, y automatiza las aprobaciones y los pagos, todo en un mismo lugar.`,
```

File: src/languages/es.ts [hunk 14]
```diff
@@ -2950,6 +2955,10 @@ export default {
                 disconnectText: 'Para desactivar la contabilidad, desconecta tu conexión contable del espacio de trabajo.',
                 manageSettings: 'Gestionar la configuración',
             },
+            rules: {
+                title: 'Reglas',
+                subtitle: 'Configura cuándo se requieren recibos, marca gastos altos y más.',
```
Comment (pecanoro) [inline src/languages/es.ts:2960]:
```suggestion
                subtitle: 'Configura cuándo se exigen los recibos, marca los gastos elevados y mucho más.',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by WojtekBoman** [comment]
@marcaaron @pecanoro I updated translations and renamed the beta :)

**Review by pecanoro — DISMISSED** [review]

**Comment by marcaaron** [comment]
Awesome thanks @pecanoro again for checking the translations 🙇 

Just gonna pop a quick hold on here as we need the beta code to get deployed, but everything LGTM @WojtekBoman.

@mkhutornyi looks like you started testing here are you going to review this PR and/or finish testing?

**Review by marcaaron — DISMISSED** [review]

**Review by mkhutornyi — APPROVED** [review]

**Review by marcaaron — APPROVED** [review]
LGTM. The beta PR should be on production now so pulling out all the stops here! Nice changes overall and also thank you @mkhutornyi for the review and testing 🙇

**Pull request merged**